unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1295,31 +1295,53 @@ bool shouldRenderImageSynthesis
                 Camera camera = thirdPartyCameras.ToArray()[i];
                 cMetadata.thirdPartyCameraId = i;
 
-                cMetadata.position = camera.gameObject.transform.position;
-                cMetadata.rotation = camera.gameObject.transform.eulerAngles;
-
-                //agent relative third party camera metadata here
-                //if the camera is a child of the base agent, then return local space values
-                if (camera.GetComponentInParent<BaseAgentComponent>()) {
-                    cMetadata.agentPositionRelativeThirdPartyCameraPosition = camera
-                        .gameObject
-                        .transform
-                        .localPosition;
-                    cMetadata.agentPositionRelativeThirdPartyCameraRotation = camera
-                        .gameObject
-                        .transform
-                        .localEulerAngles;
+                //to be depracated at some point, will be replaced by more descriptive worldRelativeThirdPartyCamera....
+                cMetadata.position = camera.transform.position;
+                cMetadata.rotation = camera.transform.eulerAngles;
+                //currently redundant data as it is the same as metadata.position/rotation, but more descriptive naming convention
+                cMetadata.worldRelativeThirdPartyCameraPosition = camera.transform.position;
+                cMetadata.worldRelativeThirdPartyCameraRotation = camera.transform.eulerAngles;
+
+                //grab this for parent or agent relative stuff
+                var worldSpaceCameraRotationAsQuaternion = camera.transform.rotation;
+
+                //this may be the same info as the agentPositionRelative values since ThirdPartyCameras often are children of the agent
+                //but in cases where a third party camera is attached to an arm joint or the like, this can be useful
+                if (camera.transform.parent != null) {
+                    cMetadata.parentObjectName = camera.transform.parent.name;
+
+                    cMetadata.parentPositionRelativeThirdPartyCameraPosition =
+                        camera.transform.parent.InverseTransformPoint(camera.transform.position);
+
+                    //get third party camera rotation as quaternion in world space
+                    var parentSpaceCameraRotationAsQuaternion =
+                        Quaternion.Inverse(camera.transform.parent.rotation)
+                        * worldSpaceCameraRotationAsQuaternion;
+                    cMetadata.parentPositionRelativeThirdPartyCameraRotation =
+                        parentSpaceCameraRotationAsQuaternion.eulerAngles;
```

Comment:
Similar to position, all you need to do to performantly get the immediate parent-relative eulerAngles is this:
`cMetadata.parentPositionRelativeThirdPartyCameraPosition = camera.transform.localEulerAngles;`

Reply:
Similar to above, I prefer the clarity explicitly performing this transformation, but I think @AlvaroHG and @Lucaweihs should definitely weigh in here, I'd like their opinion on this.

Reply:
I think since the parameter is named `localPosition` `localEulerAngles` it makes it explicit so I would avoid making an extra inverse transformation, even though optimized inverse matrix transform can be one of the slower operations

Reply:
updated with b86121869dbce2765a70b4893b4cca5072f5262e